File: 2024/day02/java/Main.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Main {
    static String readInput(String path) throws IOException {
        return Files.readString(Path.of(path)).strip();
    }

    static String part1(String data) {
        return "unimplemented";
    }

    static String part2(String data) {
        return "unimplemented";
    }

    public static void main(String[] args) throws IOException {
        String data = readInput("../input.txt");
        System.out.println("Part 1: " + part1(data));
        System.out.println("Part 2: " + part2(data));
    }
}

File: 2024/day02/rust/src/main.rs
use std::fs;

fn read_input(path: &str) -> String {
    fs::read_to_string(path)
        .expect("Failed to read input file")
        .trim()
        .to_string()
}

fn part1(data: &str) -> String {
    let _ = data;
    String::from("unimplemented")
}

fn part2(data: &str) -> String {
    let _ = data;
    String::from("unimplemented")
}

fn main() {
    let data = read_input("../input.txt");
    println!("Part 1: {}", part1(&data));
    println!("Part 2: {}", part2(&data));
}
